apps/opik-backend/src/main/java/com/comet/opik/domain/DatasetItemDAO.java

```diff
@@ -155,10 +167,12 @@ INSERT INTO dataset_items (
 
     private static final String SELECT_DATASET_ITEMS_COUNT = """
                 SELECT
-                    count(id) AS count
+                    count(id) AS count,
+                    arrayDistinct(arrayFlatten(groupArray(mapKeys(input_data)))) AS columns
```

Comment:
I forgot mentioning this in my previous revision. There might be a lot of data within the dataset for this columns. We need to consider doing this just for the page it's been queried.

Reply:
Discussed that this was rised up with FE team, but they need the columns for all dataset items, not just for the page. Performance might be a concern at some point, but we can handle it to begin with as dataset items shouldn't have much cardinality within a dataset.